test/refactoring/ConvertAsync/basic.swift

```diff
@@ -177,29 +354,81 @@ func alreadyThrows(completion: (String) -> Void) throws { }
+// RUN: %refactor-check-compiles -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix C-CONVENTION %s
 func cConvention(completion: @convention(c) () -> Void) { }
 // C-CONVENTION: func cConvention() async { }
 
-// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix VOID-HANDLER %s
+// RUN: %refactor-check-compiles -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix VOID-HANDLER %s
 func voidCompletion(completion: (Void) -> Void) {}
+// VOID-HANDLER: {
+// VOID-HANDLER-NEXT: detach {
+// VOID-HANDLER-NEXT: await voidCompletion()
+// VOID-HANDLER-NEXT: completion(())
+// VOID-HANDLER-NEXT: }
+// VOID-HANDLER-NEXT: }
 // VOID-HANDLER: func voidCompletion() async {}
 
-// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix VOID-AND-ERROR-HANDLER %s
+// RUN: %refactor-check-compiles -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix VOID-AND-ERROR-HANDLER %s
 func voidAndErrorCompletion(completion: (Void?, Error?) -> Void) {}
+// VOID-AND-ERROR-HANDLER: {
+// VOID-AND-ERROR-HANDLER-NEXT: detach {
+// VOID-AND-ERROR-HANDLER-NEXT: do {
+// VOID-AND-ERROR-HANDLER-NEXT: try await voidAndErrorCompletion()
+// VOID-AND-ERROR-HANDLER-NEXT: completion((), nil)
+// VOID-AND-ERROR-HANDLER-NEXT: } catch {
+// VOID-AND-ERROR-HANDLER-NEXT: completion(nil, error)
+// VOID-AND-ERROR-HANDLER-NEXT: }
+// VOID-AND-ERROR-HANDLER-NEXT: }
+// VOID-AND-ERROR-HANDLER-NEXT: }
 // VOID-AND-ERROR-HANDLER: func voidAndErrorCompletion() async throws {}
 
-// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix TOO-MUCH-VOID-AND-ERROR-HANDLER %s
+// RUN: %refactor-check-compiles -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix TOO-MUCH-VOID-AND-ERROR-HANDLER %s
 func tooMuchVoidAndErrorCompletion(completion: (Void?, Void?, Error?) -> Void) {}
+// TOO-MUCH-VOID-AND-ERROR-HANDLER: {
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: detach {
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: do {
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: try await tooMuchVoidAndErrorCompletion()
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: completion((), (), nil)
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: } catch {
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: completion(nil, nil, error)
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: }
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: }
+// TOO-MUCH-VOID-AND-ERROR-HANDLER-NEXT: }
 // TOO-MUCH-VOID-AND-ERROR-HANDLER: func tooMuchVoidAndErrorCompletion() async throws {}
 
+// RUN: %refactor-check-compiles -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix VOID-PROPER-AND-ERROR-HANDLER %s
+func tooVoidProperAndErrorCompletion(completion: (Void?, String?, Error?) -> Void) {}
```

Comment:
Ah, nice catch... perhaps we should skip void in the return type entirely? @hamishknight

Reply:
You mean instead of `-> (Void, String)` have it as `-> String`? We could handle that case, though it seems unlikely the user will ever write code like this in the first place

Reply:
Yeah, probably fine not to worry about it.

Reply:
I agree, if the user writes weird code like that it’s not our job to improve it while refactoring.